src/test/java/uk/gov/companieshouse/items/orders/api/controller/CertificateItemsControllerUnitTest.java

```diff
@@ -31,48 +36,64 @@
-        createdCertificateItemDTO.setId("CHS1");
-        createdCertificateItemDTO.setCompanyNumber(newCertificateItemDTO.getCompanyNumber());
-        createdCertificateItemDTO.setKind("certificate");
-        createdCertificateItemDTO.setDescriptionIdentifier("certificate");
+        final CertificateItemDTO expectedItem = new CertificateItemDTO();
+        expectedItem.setId(EXPECTED_ITEM_ID);
+        expectedItem.setCompanyNumber(newItem.getCompanyNumber());
+        expectedItem.setKind("certificate");
+        expectedItem.setDescriptionIdentifier("certificate");
         final ItemCosts costs = new ItemCosts();
         costs.setDiscountApplied("1");
         costs.setIndividualItemCost("2");
         costs.setPostageCost("3");
         costs.setTotalCost("4");
-        createdCertificateItemDTO.setItemCosts(costs);
-        createdCertificateItemDTO.setItemOptions(options);
-        createdCertificateItemDTO.setPostalDelivery(true);
-        createdCertificateItemDTO.setQuantity(5);
+        expectedItem.setItemCosts(costs);
+        expectedItem.setItemOptions(options);
+        expectedItem.setPostalDelivery(true);
+        expectedItem.setQuantity(5);
 
         // When and Then
-        mockMvc.perform(post("/certificates")
+        final ResultActions outcome = mockMvc.perform(post("/certificates")
                 .contentType(MediaType.APPLICATION_JSON)
-                .content(objectMapper.writeValueAsString(newCertificateItemDTO)))
+                .content(objectMapper.writeValueAsString(newItem)))
                 .andExpect(status().isCreated())
-                .andExpect(content().json(objectMapper.writeValueAsString(createdCertificateItemDTO)))
+                .andExpect(content().json(objectMapper.writeValueAsString(expectedItem)))
                 .andExpect(jsonPath("$.item_options.cert_inc", is(true)))
                 .andExpect(jsonPath("$.item_options.cert_shar", is(true)))
                 .andExpect(jsonPath("$.item_options.cert_dissliq", is(false)))
-                .andExpect(jsonPath("$.postal_delivery", is(true))).andDo(MockMvcResultHandlers.print());
+                .andExpect(jsonPath("$.postal_delivery", is(true)))
+                .andDo(MockMvcResultHandlers.print());
 
         // Then
-        // TODO PCI-324 Verify can retrieve equivalent entity
+        assertItemSavedCorrectly(EXPECTED_ITEM_ID);
+    }
 
+    /**
+     * Verifies that the item assumed to have been created by the create item POST request can be retrieved
+     * from the database using its expected ID value.
+     * @param expectedItemId the expected ID of the newly created item
+     */
+    private void assertItemSavedCorrectly(final String expectedItemId) {
+        final Optional<CertificateItem> retrievedCertificateItem = repository.findById(expectedItemId);
+        assertThat(retrievedCertificateItem.isPresent(), is(true));
+        assertThat(retrievedCertificateItem.get().getId(), is(expectedItemId));
```

Comment:
Is this using the embedded Mongo (sorry if I have misread the code changes)? We usually test a single unit (class) by mocking out interactions with other classes using Mockito in the unit tests.

Reply:
Yes, you could argue this is an integration test because it examines what ends up in the database. The name was made into `...UnitTest` to reflect some distinction we (will?) make in the CI tests.

Reply:
Can you change CertificateItemsControllerTest to CertificateItemsControllerIntegrationTest to make it clear it's not a unit test please?

Reply:
Will rename this test class.